frontend/src/components/maps/leaflet/LeafletDraw/LeafletDraw.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import { useEffect } from 'react';
+import { useLeaflet } from 'react-leaflet';
+import * as L from 'leaflet';
+import { Feature, FeatureCollection } from 'geojson';
+import './LeafletDraw.scss';
+require('leaflet-draw');
+
+interface ILeafletDrawProps {
+  /**
+   * Current value of the editable feature group
+   */
+  value: FeatureCollection;
+  /**
+   * Triggered when layers in the editable featureGroup have been edited and saved
+   */
+  onChange: (featureCollection: FeatureCollection) => void;
+  /**
+   * Triggered when a new feature (Polygon) has been created.
+   */
+  onCreate: (feature: Feature) => void;
+  /**
+   * Triggered when layers have been removed (and saved) from the editable featureGroup.
+   */
+  onDelete?: (featureCollection: FeatureCollection) => void;
+  /**
+   * Disable/hide the polygon draw toolbar
+   */
+  readonly?: boolean;
+  /**
+   * Only disable adding new polygons to the editable feature group
+   */
+  canAdd?: boolean;
+  /**
+   * Color of the polygon boundaries
+   */
+  color?: string;
+}
+
+const featureGroup = new L.FeatureGroup();
+
+/**
+ * A component to draw polygons on a map
+ * @example ./LeafletDraw.md
+ */
+const LeafletDraw: React.FC<ILeafletDrawProps> = ({
+  onChange,
+  onCreate,
+  value,
+  onDelete,
+  readonly,
+  canAdd,
+  color = '#fcba19',
+}) => {
+  const { map } = useLeaflet();
+
+  useEffect(() => {
+    if (!!map) {
+      const drawControl = new (L.Control as any).Draw({
+        position: 'topright',
+        draw: !readonly && {
+          rectangle: canAdd,
+          polygon: canAdd,
+          circle: false,
+          circlemarker: false,
+          marker: false,
+          polyline: false,
+        },
+        edit: {
+          featureGroup,
+          edit: !readonly,
+          remove: !readonly,
+        },
+      });
+      map.addControl(drawControl);
+    }
+  }, [map, readonly, canAdd]);
+
+  useEffect(() => {
+    if (!!map) {
+      map.removeEventListener((L as any).Draw.Event.CREATED);
```

Comment:
what's going on with these event listeners?

Reply:
might make more sense to put these in the useEffects cleanup then like this, because presumably this isn't required initially, but is required after the component is cleaned up.

Reply:
Thats a great idea